tests/integration/utils/comparisons.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from itertools import islice
+from typing import Any, Union
+
+import numpy as np
+
+
+def allclose(
+    a: Any,  # numpy or cunumeric array-like
+    b: Any,  # numpy or cunumeric array-like
+    rtol: float = 1e-5,
+    atol: float = 1e-8,
+    equal_nan: bool = False,
+    *,
+    diff_limit: Union[int, None] = 5,  # None means no limit at all
```

Comment:
Not `Optional[int]`?

Reply:
I don't love `Optional` at all, it's semantics are pretty muddled. Any kwarg with a default value is absolutelt "optional" in the practical everyday sense, but not `Optional` unless is happens to also take None. I tend to avoid it and as soon a we have min py 3.8 I'd propose to use e.g. `int | None` everywhere for explicitness while avoiding an import.